### PR TITLE
server: implement user creation route

### LINKED ISSUE #67: Implements route for user creation

### PR DESCRIPTION
**Pull request by BourgoisMickael**
Closes #67

### CODE REVIEW

File: server/src/action/user/create.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        .isLength({ min: 1, max: 255 }).withMessage('length must be between 1 and 255')
+        .customSanitizer(capitalize),
+      body('lastname')
+        .exists({ checkNull: true }).withMessage('required field')
+        .isString().withMessage('must be a string')
+        .trim()
+        .isLength({ min: 1 }).withMessage('length must be greater than 1')
+        .customSanitizer(uppercase),
+      body('password')
+        .exists({ checkNull: true }).withMessage('required field')
```
Comment (Arignir) [inline server/src/action/user/create.js:35]: The `bcrypt` implementation takes input of 72 characters maximum. Any longer password will be truncated.

I think we should limit the password's length to 72. I'm not a fan of that (i like password to have an infinite length), but unless you're ready to switch to something that also accepts infinite input i really  do not wanna give a false sensation of security to the user.

**Review by Arignir — CHANGES_REQUESTED** [review]

File: server/src/action/user/create.js (hunk 1 above)
Comment (BourgoisMickael) [reply server/src/action/user/create.js:35]: 72 is more than enough, we don't need infinite length password we are not a nuclear plant...
To ensure the password is strong enough, I think I'll check that it has at least 8 characters and it contains a number, a lowercase and an uppercase

Comment (Arignir) [reply server/src/action/user/create.js:35]: I'm not saying it's not enough, i'm suggesting a hard limit that prevents user creation if the password is longer.

Comment (BourgoisMickael) [reply server/src/action/user/create.js:35]: yes it will be set

Comment (Arignir) [reply server/src/action/user/create.js:35]: (server side)

**Comment by Arignir** [comment]
Not merging until there's a server side fix to ensure password are < 72 bytes.

We could use only client-side verification but that's bad practice, and we could postpone that and add an issue but it's literally right here, in our hand, and not doing it now would by entirely because of laziness, which is also bad.

Btw, to save a round of reviews, remember to count the number of *bytes* (and not characters) there is in the string, as that's what matters here.

**Comment by BourgoisMickael** [comment]
Forgot to push

**Review by Arignir — APPROVED** [review]

**Pull request merged**